app/controllers/listings.controller.js

```diff
@@ -13,4 +13,41 @@ exports.list = function (req, res, next) {
             }
         }).populate('user');
     }
-};
\ No newline at end of file
+};
+
+exports.createListing = function(req, res, next) {
+    var name = (req.body.name) ? req.body.name.trim() : '';
+    var price = req.body.price;
+    var description = (req.body.description) ? req.body.description.trim() : '';
+    var imageUrl = (req.body.imageUrl) ? req.body.imageUrl.trim() : '';
+    console.log(description);
+    var user = authentication.getUserFromToken(req);
+
+    //authenticate
+    if (!user) {
+        res.status(401).send('unauthorized');
+    } else if (name === '') {
+        res.status(400).send({successful: false, text: 'Please provide a name.'});
+    } else {
+        new Listing({
+            name: name,
+            description: description,
+            price: price,
+            imageUrl: imageUrl,
+            user: user._id
+
+        }).save(function(err) {
+            if (err) {
+                console.log('fail');
+                res.status(500).send({successful: false, text: 'Failed to create listing:.' + name});
```

Comment:
This response gets triggered if I provide a badly formatted price that does not parse correctly. We should probably check for this and send a 400 for this case